tests/Features/ReturnTypes/CustomEloquentBuilderTest.php

```diff
@@ -96,4 +101,15 @@ public function type(string $type): CustomBuilder
     {
         return $this->where(['type' => $type]);
     }
+
+    /**
+     * @param string[] $categories
+     *
+     * @return CustomBuilder
+     * @phpstan-return CustomBuilder<TModelClass>
```

Comment:
I think other phpdoc parsing software understand `CustomBuilder<TModelClass>`

Reply:
PHPStorm was complaining about it for me.

Reply:
I see. Generics! https://plugins.jetbrains.com/plugin/12754-php-generics

Reply:
Oh, cool. But I don't want to install a plugin for that. PHPStorm will eventually support it.